msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -1500,58 +1498,58 @@ private AccountRecord selectAccountRecordForTokenRequest(
-                            null
-                    );
-
-                    throw new MsalClientException(errMsg);
-                }
-
-                tenantIdNameOrAlias = paths.get(0);
+            if (!isUuid && !isHomeTenantEquivalent(tenantId)) {
+                tenantId = ((AzureActiveDirectoryAuthority) authority).getAudience().getTenantUuidForAlias();
             }
 
-            final IAccount accountForRequest;
+            IAccount accountForRequest;
 
-            if (isHomeTenantEquivalent(tenantIdNameOrAlias)
-                    || isAccountHomeTenant(multiTenantAccount.getClaims(), tenantIdNameOrAlias)) {
+            if (isHomeTenantEquivalent(tenantId)
+                    || isAccountHomeTenant(multiTenantAccount.getClaims(), tenantId)) {
                 accountForRequest = multiTenantAccount;
             } else {
-                accountForRequest = multiTenantAccount.getTenantProfiles().get(tenantIdNameOrAlias);
+                accountForRequest = multiTenantAccount.getTenantProfiles().get(tenantId);
             }
 
+            // If we are not able to get the AccountRecord for the requested tenanted authority,
+            // it means the user is attempting to get a token for a new tenant for the existing IAccount.
+            // For silent request, use home account if available or any of the tenant profiles and pass it
+            // along as accountForRequest. Controllers will use this account to get refresh token and acquire
+            // a new access token for the requested tenant.
             if (null == accountForRequest) { // We did not find a profile to use
                 final boolean isSilent = tokenParameters instanceof AcquireTokenSilentParameters;
-
                 if (isSilent) {
-                    validateClaimsExistForTenant(tenantIdNameOrAlias, null);
-                } else {
-                    // We didn't find an Account but the request is interactive so we'll
-                    // return null and let the user sort it out.
-                    return null;
+                    if(rootAccount.getClaims() != null){
+                        accountForRequest = rootAccount;
+                    }else {
+                        for(ITenantProfile tenantProfile : multiTenantAccount.getTenantProfiles().values()){
+                            if(tenantProfile.getClaims() != null){
+                                accountForRequest = tenantProfile;
+                                break;
+                            }
+                        }
+                    }
                 }
             }
+            // We should never hit this flow as IAccount should home profile or atleast one tenant profile on it.
+            if(accountForRequest ==  null){
+                Logger.warnPII(TAG,
+                        "No account record found for IAccount with request tenantId: " + tenantId
+                );
+                throw new ClientException(
```

Comment:
just wondering: I see this changed from `MsalClientException` to `ClientException` -- how come?

Reply:
Since the callback we are using here to post exception is `CommandCallback` , it converts all the `common` exception to `MsalExceptions`. `getTenantUuidForAlias` here would throw a `ClientException` , so made this `ClientException` to be consistent and avoid have multiple exceptions declared in throws